coding_systems/bnf/import_data.py

```diff
@@ -1,33 +1,45 @@
 import csv
-import glob
-import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
 
-from django.db import transaction
+import structlog
 
-from .models import TYPES, Concept
+from coding_systems.base.import_data_utils import CodingSystemImporter
+from coding_systems.bnf.models import TYPES, Concept
 
+logger = structlog.get_logger()
 
-def import_data(release_dir):
-    paths = glob.glob(os.path.join(release_dir, "*.csv"))
-    assert len(paths) == 1
-    path = paths[0]
 
-    records = {type: set() for type in TYPES}
+def import_data(release_zipfile, release_name, valid_from, import_ref=None):
+    with TemporaryDirectory() as tempdir:
+        release_zip = ZipFile(release_zipfile)
+        logger.info("Extracting", release_zip=release_zip.filename)
+        release_zip.extractall(path=tempdir)
+        paths = list(Path(tempdir).glob("*.csv"))
+        assert (
+            len(paths) == 1
+        ), f"Expected 1 and only one .csv file (found {len(paths)})"
+        path = paths[0]
 
-    with open(path) as f:
-        for r in csv.DictReader(f):
-            parent_code = None
-            for type in TYPES:
-                name = r[f"BNF {type}"]
-                code = r[f"BNF {type} Code"]
-                if "DUMMY" not in name:
-                    records[type].add((code, name, parent_code))
-                    parent_code = code
+        with CodingSystemImporter(
+            "bnf", release_name, valid_from, import_ref
+        ) as database_alias:
+            records = {type: set() for type in TYPES}
+            with open(path) as f:
+                for r in csv.DictReader(f):
+                    parent_code = None
+                    for type in TYPES:
+                        name = r[f"BNF {type}"]
+                        code = r[f"BNF {type} Code"]
+                        if "DUMMY" not in name:
+                            records[type].add((code, name, parent_code))
+                            parent_code = code
```

Comment:
It might be clearer if everything above this point was moved out of the `with CodingSystemImporter` context manager, since (afaict) it doesn't need the context.  It would also let us dedent the `with CodingSystemImporter` line and everything below it, because it would have finished using `tempdir`.